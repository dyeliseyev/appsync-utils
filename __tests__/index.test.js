/* 
  * Helpers definition from https://docs.aws.amazon.com/appsync/latest/devguide/dynamodb-helpers-in-util-dynamodb-js.html
*/

const { checkValid } = require("./helpers.js");


describe("dynamodb helpers", () => {
  describe("toDynamoDB", () => {
    test("string", async () => {
      await checkValid(`util.dynamodb.toDynamoDB("test")`);
    });
    test("number", async () => {
      await checkValid(`util.dynamodb.toDynamoDB(12345)`);
    });
    test("boolean", async () => {
      await checkValid(`util.dynamodb.toDynamoDB(true)`);
    });
  });

  test("toString", async () => {
    await checkValid(`util.dynamodb.toString("test")`);
  });

  test("toStringSet", async () => {
    await checkValid(`util.dynamodb.toStringSet(["foo", "bar", "baz"])`);
  });

  test("toNumber", async () => {
    await checkValid(`util.dynamodb.toNumber(12345)`);
  });

  test("toNumberSet", async () => {
    await checkValid(`util.dynamodb.toNumberSet([1, 23, 4.56])`);
  });

  test("toBinary", async () => {
    await checkValid(`util.dynamodb.toBinary("foo")`);
  });

  test("toBinarySet", async () => {
    await checkValid(`util.dynamodb.toBinarySet(["foo", "bar", "baz"])`);
  });

  test("toBoolean", async () => {
    await checkValid(`util.dynamodb.toBoolean(true)`);
  });

  test("toNull", async () => {
    await checkValid(`util.dynamodb.toNull()`);
  });

  test("toList", async () => {
    await checkValid(`util.dynamodb.toList(["foo", 123, {bar: "baz" }])`);
  });

  test("toMap", async () => {
    await checkValid(`util.dynamodb.toMap({ "foo": "bar", "baz": 1234, "beep": ["boop"] })`);
  });

  test("toMapValues", async () => {
    await checkValid(`util.dynamodb.toMapValues({ "foo": "bar", "baz": 1234, "beep": ["boop"] })`);
  });

  describe("s3 objects", () => {
    test("three parameter function", async () => {
      await checkValid(`util.dynamodb.toS3Object("foo", "bar", "baz")`);
    });
    test("four parameter function", async () => {
      await checkValid(`util.dynamodb.toS3Object("foo", "bar", "baz", "beep")`);
    });
    test.skip("fromS3ObjectJson", async () => {
      await checkValid(`util.dynamodb.fromS3ObjectJson({ "S" : "{ \"s3\" : { \"key\" : \"foo\", \"bucket\" : \"bar\", \"region\" : \"baz\", \"version\" = \"beep\" } }" })`);
    });
  });
});
