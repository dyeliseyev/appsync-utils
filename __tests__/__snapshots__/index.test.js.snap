// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamoDB module functions get 1`] = `
{
  "key": {
    "id": {
      "S": "id",
    },
  },
  "operation": "GetItem",
}
`;

exports[`DynamoDB module functions put 1`] = `
{
  "attributeValues": {
    "value": {
      "N": 10,
    },
  },
  "key": {
    "id": {
      "S": "abc",
    },
  },
  "operation": "PutItem",
}
`;

exports[`DynamoDB module functions remove 1`] = `
{
  "key": {
    "id": {
      "S": "test",
    },
  },
  "operation": "DeleteItem",
}
`;

exports[`DynamoDB module functions scan 1`] = `
{
  "limit": 10,
  "nextToken": "abc",
  "operation": "Scan",
}
`;

exports[`DynamoDB module functions sync 1`] = `undefined`;

exports[`DynamoDB module functions update add 1`] = `
{
  "key": {
    "id": {
      "S": "test",
    },
  },
  "operation": "UpdateItem",
  "update": {
    "expression": "SET #expName_1 = :expValue_1",
    "expressionNames": {
      "#expName_1": "age",
    },
    "expressionValues": {
      ":expValue_1": {
        "N": 10,
      },
    },
  },
}
`;

exports[`DynamoDB module functions update append 1`] = `
{
  "key": {
    "id": {
      "S": "test",
    },
  },
  "operation": "UpdateItem",
  "update": {
    "expression": "SET #expName_1 = list_append(#expName_1, :expValue_1)",
    "expressionNames": {
      "#expName_1": "values",
    },
    "expressionValues": {
      ":expValue_1": {
        "L": [
          {
            "N": 1,
          },
          {
            "N": 2,
          },
          {
            "N": 3,
          },
        ],
      },
    },
  },
}
`;

exports[`DynamoDB operations add 1`] = `
{
  "type": "OPERATION_ADD",
  "value": 10,
}
`;

exports[`DynamoDB operations append 1`] = `
{
  "items": [
    1,
    2,
    3,
  ],
  "type": "OPERATION_APPEND",
}
`;

exports[`DynamoDB operations decrement 1`] = `
{
  "by": 10,
  "type": "OPERATION_DECREMENT",
}
`;

exports[`DynamoDB operations increment 1`] = `
{
  "by": 10,
  "type": "OPERATION_INCREMENT",
}
`;

exports[`DynamoDB operations prepend 1`] = `
{
  "items": [
    1,
    2,
    3,
  ],
  "type": "OPERATION_PREPEND",
}
`;

exports[`DynamoDB operations replace 1`] = `
{
  "type": "OPERATION_REPLACE",
  "value": {
    "a": 10,
  },
}
`;

exports[`DynamoDB operations updateListItem 1`] = `undefined`;

exports[`dynamodb helpers s3 objects four parameter function 1`] = `
{
  "S": "{"s3":{"key":"foo","bucket":"bar","region":"baz","version":"beep"}}",
}
`;

exports[`dynamodb helpers s3 objects fromS3ObjectJson 1`] = `undefined`;

exports[`dynamodb helpers s3 objects three parameter function 1`] = `
{
  "S": "{"s3":{"key":"foo","bucket":"bar","region":"baz"}}",
}
`;

exports[`dynamodb helpers toBinary 1`] = `
{
  "B": "foo",
}
`;

exports[`dynamodb helpers toBinarySet 1`] = `
{
  "BS": [
    "foo",
    "bar",
    "baz",
  ],
}
`;

exports[`dynamodb helpers toBoolean 1`] = `
{
  "BOOL": true,
}
`;

exports[`dynamodb helpers toDynamoDB boolean 1`] = `
{
  "BOOL": true,
}
`;

exports[`dynamodb helpers toDynamoDB number 1`] = `
{
  "N": 12345,
}
`;

exports[`dynamodb helpers toDynamoDB string 1`] = `
{
  "S": "test",
}
`;

exports[`dynamodb helpers toList 1`] = `
{
  "L": [
    {
      "S": "foo",
    },
    {
      "N": 123,
    },
    {
      "M": {
        "bar": {
          "S": "baz",
        },
      },
    },
  ],
}
`;

exports[`dynamodb helpers toMap 1`] = `
{
  "M": {
    "baz": {
      "N": 1234,
    },
    "beep": {
      "L": [
        {
          "S": "boop",
        },
      ],
    },
    "foo": {
      "S": "bar",
    },
  },
}
`;

exports[`dynamodb helpers toMapValues 1`] = `
{
  "baz": {
    "N": 1234,
  },
  "beep": {
    "L": [
      {
        "S": "boop",
      },
    ],
  },
  "foo": {
    "S": "bar",
  },
}
`;

exports[`dynamodb helpers toNull 1`] = `
{
  "NULL": null,
}
`;

exports[`dynamodb helpers toNumber 1`] = `
{
  "N": 12345,
}
`;

exports[`dynamodb helpers toNumberSet 1`] = `
{
  "NS": [
    1,
    23,
    4.56,
  ],
}
`;

exports[`dynamodb helpers toString 1`] = `
{
  "S": "test",
}
`;

exports[`dynamodb helpers toStringSet 1`] = `
{
  "SS": [
    "foo",
    "bar",
    "baz",
  ],
}
`;
